core/API/CORSHandler.php

```diff
@@ -18,40 +20,85 @@ class CORSHandler
      */
     protected $domains;
 
+    private $logger;
+
     public function __construct()
     {
         $this->domains = Url::getCorsHostsFromConfig();
+        $this->logger = StaticContainer::get(LoggerInterface::class);
+
     }
 
+    /**
+     * This method is set header for the request to pass browser cross domain checks.
+     * By default, it allow from all hosts
```

Comment:
Could you maybe try to improve that? I actually wouldn't know what the method is doing only by reading this comment.